community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/UsingAcceptanceTest.scala

```diff
@@ -225,4 +225,23 @@ class UsingAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTestSup
     intercept[SyntaxException](
       executeWithAllPlanners("MATCH n-->() USING SCAN n:Person return n"))
   }
+
+  test("when failing to support all hints we should provide an understandable error message") {
+    // GIVEN
+    graph.createIndex("LocTag", "id")
+
+    // WHEN
+    val query = """CYPHER planner=greedy MATCH (t1:LocTag {id:1642})-[:Child*0..]->(:LocTag)
+                  |     <-[:Tagged]-(s1:Startup)<-[r1:Role]-(u:User)
+                  |     -[r2:Role]->(s2:Startup)-[:Tagged]->(:LocTag)
+                  |     <-[:Child*0..]-(t2:LocTag {id:1642})
+                  |USING INDEX t1:LocTag(id)
+                  |USING INDEX t2:LocTag(id)
+                  |RETURN count(u)""".stripMargin
+
+
+    val error = intercept[HintException](innerExecute(query))
```

Comment:
Why do we expect to fail on this query? Isn't the greedy planner able to support both hints?